### PR TITLE
Restricted positionFacebookBadge function argument to accept only one data type

### LINKED ISSUE #296: Content_Script Revisions
- [x] Refactor `detectFacebookOnPage`/`PATTERN` Blocks to reuse code better 
- [x] Revise `positionFacebookBadge` function to assume first arg is always a selector, and remove object detection

### PR DESCRIPTION
**Pull request by maxxcrawford**
Closed #296 

- Added Medium targeted support. (Doesn't work as it doesn't block any known trackers on the page)

### CODE REVIEW

File: src/content_script.js [hunk 1]
```diff
@@ -26,9 +26,15 @@ const SHARE_PATTERN_DETECTION_SELECTORS = [
...
   "[href*='facebook.com/sharer']", // Buzzfeed
-  "[aria-label*='share on facebook']" // MSN
+  "[data-bfa-network*='facebook']", // Buzzfeed Mini Share
+  "[aria-label*='share on facebook']", // MSN
+  "[title='Share on Facebook']" // Medium
 ];
 
+// const SHARE_DISABLE_DETECTION_SELECTORS = [
+//   "[class*='js-share-button--facebook']" // Buzzfeed
+// ];
```
Comment (groovecoder) [inline src/content_script.js]: Need to keep this? Or can we delete it completely?

File: src/content_script.js [hunk 2]
```diff
@@ -222,9 +228,15 @@ function addFacebookBadge (target, badgeClassUId, socialAction) {
       document.querySelector(".fbc-has-badge.js-fbc-prompt-active").classList.remove("js-fbc-prompt-active");
       e.target.parentElement.parentNode.parentNode.classList.remove("active");
     });
+  } else if (socialAction === "share") {
+    target.addEventListener("click", (e) => {
+      e.preventDefault();
+      return false;
```
Comment (groovecoder) [inline src/content_script.js]: Should we add explicit `return` statement in the `socialAction === "login"` block? Should we have a default `return` outside of the `if` blocks?

File: src/content_script.js (hunk 1 above)
Comment (maxxcrawford) [reply src/content_script.js]: Migrated to a TO-DO.

File: src/content_script.js (hunk 2 above)
Comment (maxxcrawford) [reply src/content_script.js]: Removed return false for now.

**Comment by maxxcrawford** [comment]
@groovecoder Want me to catch this up with `add-sites-to-FBC` first?

**Pull request merged**